### PR TITLE
refactor: fixes for security and types

### PR DESCRIPTION
**Pull request by julianmrodri**
* Applies refactors based on Security suggestions
* Modifies all capacity/size variables to `uint64`
* Bumps solidity version and contracts versions to latest (suggested by Security)
* Adds license in each contract (required by Solidity `0.6.12`)

### CODE REVIEW

File: contracts/Staking.sol [hunk 1]
```diff
@@ -78,11 +81,11 @@ contract Staking is Ownable {
             amount = msg.value;
             tokenAddress = address(0);
         } else {
-            require(IERC20(tokenAddress).transferFrom(msg.sender, address(this), amount), "Staking: could not transfer tokens");
```
Comment (nduchak) [inline contracts/Staking.sol:81]: Why do you remove this require?

**Review by nduchak — APPROVED** [review]
Nice job 👍

File: contracts/Staking.sol (hunk 1 above)
Comment (julianmrodri) [reply contracts/Staking.sol:81]: The security recommendation was:  **"in function unstake(): add require() around IERC20 transfer() call as performed in stakeFor() for example. Also, consider using SafeERC20 [refs: https://soliditydeveloper.com/safe-erc20, https://docs.openzeppelin.com/contracts/2.x/api/token/erc20#SafeERC20"**

I replaced the validation by using a contract provided by OZ named `SafeERC20` which is used as a library and already has this validation. I guess its just a different way of doing it.

Comment (julianmrodri) [reply contracts/Staking.sol:81]: Then I did the same thing for `stake` just to be consistent.

Comment (nduchak) [reply contracts/Staking.sol:81]: Clear, but then should we also use safeTransfer in the Storage Manager contract?

Comment (julianmrodri) [reply contracts/Staking.sol:81]: I think the StorageManager contract cannot take any more code. Its quite big already, thats why i kept the previous approach in this one. No margin for sugar!

**Pull request merged**